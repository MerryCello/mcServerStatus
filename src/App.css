.App {
  text-align: center;
}

body {
  padding-top: 7vh;
  background-color: rgb(30,30,30);
  background-image: url(./images/dirt_bg.png);
  background-position: center;
  background-repeat: repeat;
  background-size: cover;
  color: #fcfcfc;
  font-family: "Minecraft";
  font-size: 1.4rem;
}
h1 {
  font-size: 1.4rem;
  text-shadow: 4px 4px #3e3e3e;
  padding-left: 5%;
  padding-right: 5%;
}
table {
  margin-left: 5%;
  margin-right: 5%;
}
th {
  padding-bottom: 10px;
}
th, td {
  padding-left: 10px;
  padding-right: 10px;
  padding-bottom: 15px;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.card-grid {
  display: grid;
  grid-template-columns: auto 1fr;
  padding: 3px;
}

div.card-grid:focus {
  background-color: black;
  outline: 3px solid white;
}

.card-icon {
  width: fit-content;
}

.card-icon>img {
  height: 6rem;
  width: auto;
  padding-right: 8px;
}

.server-details {
  display: flex;
  flex-direction: column;
  flex: 3 4;
  line-height: 1.5rem;
}

.card-srv-name-status {
  display: grid;
  grid-template-columns: auto auto;
  align-items: start;
  line-height: 2rem !important;
}

span.card-srv-name-status {
  justify-content: end;
  margin-left: 10px;
}

.servers-list {
  margin: 0 18% 0 18%;
  padding: 10px 12px 0 3px;
  height: 60vh;
  overflow-y: auto;
}

.servers-list-bg {
  background-color: rgba(0, 0, 0, 0.5);
  padding: 3px 0 3px 0;
}

::-webkit-scrollbar {
  width: 18px;
}

::-webkit-scrollbar-track {
  background-color: black;
}

::-webkit-scrollbar-thumb {
  background-color: #c0c0c0;
  border-right: 3px solid #808080;
  border-bottom: 3px solid #808080;
}

::-webkit-scrollbar-corner {
  background-color: transparent;
}

.button, .button:focus {
  color: #fcfcfc;
  text-shadow:4px 4px rgba(0, 0, 0, 0.5);
  background-image: url(./images/gray-texture.png);
  background-position: center;
  background-repeat: repeat;
  border-top: 3px solid #aaaaaa;
  border-left: 3px solid #aaaaaa;
  border-bottom: 6px solid #565656;
  border-right: 3px solid #565656;
  outline: 3px solid black;
  height: 7vh;
}

.button:disabled, .button:disabled:hover {
  border: none;
  background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url(./images/gray-texture.png);
  color: #9e9e9e;
  text-shadow: none;
  vertical-align: baseline;
  outline: 3px solid black;
}

.button:hover {
  outline: 3px solid white;
}

.options {
  display: grid;
  grid-template-columns: auto auto auto auto;
  margin: 3% 18% 0 18%;
  gap: 15px;
}

@media only screen and (max-width: 1000px) {
  .servers-list {
    margin: 0 5% 0 5%;
  }
  .options {
    display: grid;
    grid-template-columns: auto auto;
    margin: 3% 5% 0 5%;
  }
  body, h1 {
    font-size: 1.25rem;
  }
}

@media only screen and (max-width: 500px) {
  body {
    font-size: .75rem;
  }
  button {
    font-size: 1rem;
  }
  .card-icon>img {
    width: 15vw;
    height: auto;
    padding-right: 8px;
  }
  .card-srv-name-status {
    line-height: 1rem !important;
  }
}
