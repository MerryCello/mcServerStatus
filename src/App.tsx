import React from 'react';
import LandingPage from './pages/LandingPage';

function App() {
  return <LandingPage data-testid='LandingPage' />;
}

export default App;
