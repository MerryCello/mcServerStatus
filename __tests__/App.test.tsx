/** @jest-environment jsdom */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { test, expect } from '@jest/globals';
import '@testing-library/jest-dom';
import App from '../src/App';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../src/pages/LandingPage', () => 'landing-page');

describe('App', () => {
  test('renders LandingPage by default', () => {
    const app = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const landingPage = screen.getByTestId('LandingPage');
    expect(landingPage).toBeDefined();
    expect(app.container).toMatchSnapshot();
  });
});
