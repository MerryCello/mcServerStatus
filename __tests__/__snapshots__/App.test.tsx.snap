// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders LandingPage by default 1`] = `
<div>
  <landing-page
    data-testid="LandingPage"
  />
</div>
`;
